libs/langchain-community/src/llms/friendli.ts

```diff
@@ -0,0 +1,236 @@
+import { CallbackManagerForLLMRun } from "@langchain/core/callbacks/manager";
+import {
+  type BaseLLMCallOptions,
+  type BaseLLMParams,
+  LLM,
+} from "@langchain/core/language_models/llms";
+import { GenerationChunk } from "@langchain/core/outputs";
+import { getEnvironmentVariable } from "@langchain/core/utils/env";
+import { convertEventStreamToIterableReadableDataStream } from "../utils/event_source_parse.js";
+
+/**
+ * The FriendliParams interface defines the input parameters for
+ * the Friendli class.
+ */
+export interface FriendliParams extends BaseLLMParams {
+  /**
+   * Model name to use.
+   */
+  model?: string;
+  /**
+   * Friendli personal access token to run as.
+   */
+  friendliToken?: string;
+  /**
+   * Friendli team ID to run as.
+   */
+  friendliTeam?: string;
+  /**
+   * Number between -2.0 and 2.0. Positive values penalizes tokens that have been
+   * sampled, taking into account their frequency in the preceding text. This
+   * penalization diminishes the model's tendency to reproduce identical lines
+   * verbatim.
+   */
+  frequencyPenalty?: number;
+  /**
+   * Number between -2.0 and 2.0. Positive values penalizes tokens that have been
+   * sampled at least once in the existing text.
+   * presence_penalty: Optional[float] = None
+   * The maximum number of tokens to generate. The length of your input tokens plus
+   * `max_tokens` should not exceed the model's maximum length (e.g., 2048 for OpenAI
+   * GPT-3)
+   */
+  maxTokens?: number;
+  /**
+   * When one of the stop phrases appears in the generation result, the API will stop
+   * generation. The phrase is included in the generated result. If you are using
+   * beam search, all of the active beams should contain the stop phrase to terminate
+   * generation. Before checking whether a stop phrase is included in the result, the
+   * phrase is converted into tokens.
+   */
+  stop?: string[];
+  /**
+   * Sampling temperature. Smaller temperature makes the generation result closer to
+   * greedy, argmax (i.e., `top_k = 1`) sampling. If it is `None`, then 1.0 is used.
+   */
+  temperature?: number;
+  /**
+   * Tokens comprising the top `top_p` probability mass are kept for sampling. Numbers
+   * between 0.0 (exclusive) and 1.0 (inclusive) are allowed. If it is `None`, then 1.0
+   * is used by default.
+   */
+  topP?: number;
+}
+
+/**
+ * The Friendli class is used to interact with Friendli inference Endpoint models.
+ * This requires your Friendli Token and Friendli Team which is autoloaded if not specified.
+ */
+export class Friendli extends LLM<BaseLLMCallOptions> {
+  lc_serializable = true;
+
+  static lc_name() {
+    return "Friendli";
+  }
+
+  get lc_secrets(): { [key: string]: string } | undefined {
+    return {
+      friendliToken: "FRIENDLI_TOKEN",
+      friendliTeam: "FRIENDLI_TEAM",
+    };
+  }
+
+  model = "mixtral-8x7b-instruct-v0-1";
```

Comment:
For consistency, let's set this to `llama-2-13b-chat` to match `ChatFriendli`.

Reply:
I think we do want the `instruct` here actually since this is a text completion endpoint.

I'm also ok to remove the default entirely - we've been generally moving in that direction.

Reply:
I think we need a default since `model` is required by the Friendli API.